scripts/check-android-16kb-memory-page-size-compatibility.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+set -o pipefail
+
+progname="${0##*/}"
+progname="${progname%.sh}"
+
+# usage: check_elf_alignment.sh [path to *.so files|path to *.apk]
+
+cleanup_trap() {
+  if [ -n "$tmp" ] && [ -d "$tmp" ]; then
+    rm -rf "$tmp"
+  fi
+  exit "${1:-0}"
+}
+
+usage() {
+  echo "Host side script to check the ELF alignment of shared libraries."
+  echo "Shared libraries are reported ALIGNED when their ELF regions are"
+  echo "16 KB or 64 KB aligned. Otherwise they are reported as UNALIGNED."
+  echo
+  echo "Usage: ${progname} [input-path|input-APK|input-APEX]"
+}
+
+if [ "$#" -ne 1 ]; then
+  usage
+  exit 1
+fi
+
+case "$1" in
+  --help | -h | -\?)
+    usage
+    exit 0
+    ;;
+  *)
+    dir="$1"
+    ;;
+esac
+
+if ! [ -f "$dir" ] && ! [ -d "$dir" ]; then
+  echo "Invalid file: $dir" >&2
+  exit 1
+fi
+
+if [[ "$dir" == *.apk ]]; then
+  trap 'cleanup_trap' EXIT
+
+  echo
+  echo "Recursively analyzing $dir"
+  echo
+
+  if zipalign --help 2>&1 | grep -q "\-P <pagesize_kb>"; then
+    echo "=== APK zip-alignment ==="
+    # SC2196: use grep -E instead of egrep
+    zipalign -v -c -P 16 4 "$dir" | grep -E 'lib/arm64-v8a|lib/x86_64|Verification'
+    echo "========================="
+  else
+    echo "NOTICE: Zip alignment check requires build-tools version 35.0.0-rc3 or higher."
```

Comment:
NAB: `scripts/shellUtils.sh` has utilities for printing warnings and errors with different colors